Logging Improvements

abdel opened the pull request:
As per our discussions, we will be keeping the `print` statements instead of using `logging.info` to keep the console output nice and neat! 💃 

I have made some changes to update the remaining `print` statements with any `ERROR` or `WARNING` mentions to `logging.error` and `logging.warning` respectively.

**Format:** `[FILE_NAME:LINE_NUMBER - functionName() - LEVEL] MESSAGE`

**Example:** `[run-framework.py:149 - check_args() - WARNING] A keypath has been set, but we're not running on a remote machine (arg: step_remote). It will have no effect.`

There didn't seem to be an obvious way to include the class name it self, but the file name would be a good indication of the class anyway. Let me know if you would like to alter the message format!

affogato commented on scripts/run-framework/agief_experiment/experiment.py:
```diff
@@ -113,10 +113,10 @@ def log_results_config(self, compute_node):
             except KeyError:
-                print("KeyError Exception")
-                print("WARNING: trying to access path '" + param_path + "' at config.value, but it DOES NOT exist!")
+                logging.error("KeyError Exception: trying to access path '" + 
```
This is a KeyError exception, but from the python script's perspective, it isn't serious enough to be an Error.
Could you please change to log.warning rather than log.error

Error should be something that has gone seriously wrong and serious enough to not continue.
Warning, it can continue and still be successful. This KeyError results in incomplete logging of the results, but they're still available, just not in the console. 

In light of this definition, and to be consistent, there may be other places that we should change from Error to Warning.

affogato commented on scripts/run-framework/agief_experiment/cloud.py:
```diff
@@ -101,7 +101,7 @@ def ecs_stop_task(self, task_arn):
         )
 
         if self.log:
```
Since we're using proper logging, we can now remove the `if self.log:` and all other places where we are currently manually keeping track of log level. That means also removing it as a parameter from the methods.

abdel commented:
I've made a couple of more changes today if you get a chance to re-review:
- Updated some of the `logging.error` to `logging.warning` where it made sense as discussed
- Removed the `self.log` param from a couple of classes and from `run-framework.py`
- Added a util function to map a logging level to the numerical level value from the `logging` module
-- can use the existing `args.logging` to set a level like `--logging debug`, and by default it will be set to `warning` to avoid debug statements by default (similar to how it was with self.log)
- updated `run-framework.py` to be more pylint compliant (fixing long lines, stuff like that)

affogato commented on scripts/run-framework/agief_experiment/cloud.py:
```diff
@@ -71,8 +69,7 @@ def ecs_run_task(self, task_name):
 
-        if self.log:
-            print("LOG: ", response)
+        logging.debug("LOG: " + response)
```
we don't need the LOG anymore :-)
we can remove after merge
